DataSet.rst

```diff
@@ -1,414 +1,451 @@
+in the DataSets tag of the output DataSet. In this way the original DataSets are maintained
+as "subdatasets" of the new DataSet. Merging tools should adhere to the following best practices\:
+
+- Subdatasets should be created for each input DataSet when two or more files with no subdatasets are merged or used to create a new DataSet.
+- If in the process of merging a new DataSet is created, for example, wrapping a "naked" BAM file, a subdataset should be created for those inputs.
+- In no case should a subdataset be present if it is identical to the containing DataSet.
+- Subdatasets should be preserved when two datasets with subdatasets are merged together.
+- If both have subdatasets, the lists of subdatasets should be concatenated.
+- If one has no subdatasets, a subdataset should be created for that input and added to the list of subdatasets from the other input file.
+- Subdatasets should be preserved during minor manipulations: Adding a filter, changing path absoluteness, adding or removing indices, references etc.
+- Subdatasets should be removed during substantial transformations: alignment, producing an AlignmentSet from a SubreadSet.
+- Subdatasets should be preserved during consolidation or splitting (except when splitting by subdataset).
+
+
+Consolidating
++++++++++++++
 
 Consolidating (aka Resolving) a DataSet means creating an explicit
 representation in the appropriate format with all filters applied. Here
 is consolidated version of the SubreadSet above::
 
     <?xml version="1.0" encoding="utf-8" ?>
-    <SubreadSet xmlns="http://pacificbiosciences.com/PacBioDataModel.xsd">
-        <ExternalResources>
-            <ExternalResource 
+    <pbds:SubreadSet xmlns="http://pacificbiosciences.com/PacBioDataModel.xsd">
+        <pbbase:ExternalResources>
+            <pbbase:ExternalResource
                 MetaType="SubreadFile.SubreadBamFile"
                 ResourceId="file:/mnt/path/to/subreads0_plus_subreads1.bam"/>
-        </ExternalResources>
-        <DataSetMetadata>
-            <TotalLength>5000</TotalLength>
-            <NumRecords>500</NumRecords>
-        </DataSetMetadata>
-    </SubreadSet>
+        </pbbase:ExternalResources>
+        <pbds:DataSetMetadata>
+            <pbbase:TotalLength>5000</pbbase:TotalLength>
+            <pbbase:NumRecords>500</pbbase:NumRecords>
+        </pbds:DataSetMetadata>
+    </pbds:SubreadSet>
 
 Consolidated DataSets are useful for export of a DataSet that exists
 only implicitly, e.g. by filtering multiple files. They allow the user
 to incur the IO overhead of seeking over multiple files once at the
 cost of increased disk usage.
 
 
-Labelling subsets of DataSets
-+++++++++++++++++++++++++++++
+Labelling subsets
++++++++++++++++++
 
-DataSets can contain other DataSets. These DataSets are defined relative
-to the parent DataSet, and provide the ability to label subsets of the
+DataSets can contain other DataSets. These DataSets are independent DataSets
```

Comment:
I believe the only place we are consistently using the labelled datasets is after merging data sets, preserving the originals as subdatasets. In 4.0 the merging performed by python supported subdatasets, but the merging performed by web services did not. In 4.1 both will. So yes, the tool support is spotty, but we are moving towards more regularity (of course, this spec is an important step in that direction). But maybe this is covered by the merging / union section?